Src/Support/Google.Apis.Auth/OAuth2/ProgrammaticExternalAccountCredential.cs

```diff
@@ -0,0 +1,158 @@
+        /// </summary>
+        /// <param name="tokenUrl">
+        /// The STS token exchange endpoint. Must not be null or empty.
+        /// </param>
+        /// <param name="audience">
+        /// The STS audience which contains the resource name for the workload identity pool and the provider identifier in that pool.
+        /// Must not be null or empty.
+        /// </param>
+        /// <param name="subjectTokenType">
+        /// The STS subject token type based on the <see href="https://datatracker.ietf.org/doc/html/rfc8693#section-3">OAuth 2.0 token exchange specification</see>.
+        /// Must not be null or empty.
+        /// </param>
+        /// <param name="subjectTokenProvider">
+        /// The provider of the third-party token that's included on the STS token exchange request.
+        /// Must not be null.
+        /// </param>
+        public Initializer(string tokenUrl, string audience, string subjectTokenType, ISubjectTokenProvider subjectTokenProvider) : base(
+            tokenUrl.ThrowIfNullOrEmpty(nameof(tokenUrl)),
+            audience.ThrowIfNullOrEmpty(nameof(audience)),
+            subjectTokenType.ThrowIfNullOrEmpty(nameof(subjectTokenType))) =>
+            SubjectTokenProvider = subjectTokenProvider.ThrowIfNull(nameof(subjectTokenProvider));
+
+        /// <summary>
+        /// Builds and initializer for <see cref="ProgrammaticExternalAccountCredential"/> from the given initializer.
+        /// </summary>
+        public Initializer(Initializer other)
+            : base(other.ThrowIfNull(nameof(other))) =>
+            SubjectTokenProvider = other.SubjectTokenProvider;
+
+        /// <summary>
+        /// Builds and initializer for <see cref="ProgrammaticExternalAccountCredential"/> from the given credential.
+        /// </summary>
+        public Initializer(ProgrammaticExternalAccountCredential credential)
+            : base(credential.ThrowIfNull(nameof(credential))) =>
+            SubjectTokenProvider = credential.SubjectTokenProvider;
+
+        /// <summary>
+        /// The provider of the third-party token that's included on the STS token exchange request.
+        /// Must not be null.
+        /// </summary>
+        public ISubjectTokenProvider SubjectTokenProvider { get; }
+    }
+
+    /// <summary>
+    /// The provider of the third-party token that's included on the STS token exchange request.
+    /// Must not be null.
+    /// </summary>
+    public ISubjectTokenProvider SubjectTokenProvider { get; }
+
+    /// <inheritdoc/>
+    bool IGoogleCredential.HasExplicitScopes => HasExplicitScopes;
+
+    /// <inheritdoc/>
+    bool IGoogleCredential.SupportsExplicitScopes => SupportsExplicitScopes;
+
+    /// <summary>
+    /// Builds a <see cref="ProgrammaticExternalAccountCredential"/> instace from the given initializer.
+    /// </summary>
+    public ProgrammaticExternalAccountCredential(Initializer initializer)
+        : base(initializer.ThrowIfNull(nameof(initializer))) =>
```

Comment:
Likewise, I suspect the base code checks for null.

Reply:
Nope, also starting from ServiceCredential. It makes slighty more sense to not have the checks on ExternalAccountCredential because all the initializers were internal anyway, and we were building them on the credential provider.